curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-characters-that-occur-zero-or-more-times.english.md

```diff
@@ -22,15 +22,19 @@ Create a regex <code>chewieRegex</code> that uses the <code>*</code> character t
 ```yml
 tests:
   - text: Your regex <code>chewieRegex</code> should use the <code>*</code> character to match zero or more <code>a</code> characters.
-    testString: assert(/\*/.test(chewieRegex.source), 'Your regex <code>chewieRegex</code> should use the <code>*</code> character to match zero or more <code>a</code> characters.');
-  - text: Your regex <code>chewieRegex</code> should match 16 characters.
-    testString: assert(result[0].length === 16, 'Your regex <code>chewieRegex</code> should match 16 characters.');
+    testString: assert(/\*/.test(chewieRegex.source));
+  - text: Your regex should match <code>"A"</code>.
+    testString: assert(chewieRegex.test('A'));
+  - text: Your regex should not match <code>"aaaa"</code>.
+    testString: assert(!chewieRegex.test('aaaa'));
   - text: Your regex should match <code>"Aaaaaaaaaaaaaaaa"</code>.
-    testString: assert(result[0] === 'Aaaaaaaaaaaaaaaa', 'Your regex should match <code>"Aaaaaaaaaaaaaaaa"</code>.');
+    testString: assert(result[0] === 'Aaaaaaaaaaaaaaaa');
```

Comment:
```suggestion
    testString: assert(chewieRegex.test('Aaaaaaaaaaaaaaaa'));
```

Reply:
I might be wrong about this, but I believe this change would cause incorrect answers to pass these tests. `test()` just checks whether a pattern is found anywhere in the string, so the following would work:

`let chewieRegex = /A/;`
![image](https://user-images.githubusercontent.com/2051070/54065224-1b5f4480-4261-11e9-85a3-70a488ac330e.png)

let chewieRegex = /a/;
![image](https://user-images.githubusercontent.com/2051070/54065226-29ad6080-4261-11e9-85dd-b430de678997.png)

Reply:
What we could do is change the tests to `'Aaaaaaaaaaaaaaaa'.match(chewieQuote)`, or some variation of that to make sure we're just testing the regex itself. I'm not sure if that's the best move, though, since it might give users the wrong impression about how regex works if we're hard coding some values behind the scenes.

If the user changes the first string, then plays with different regex patterns until they pass the challenge, they could be confused by the textString that says their regex should match 16 characters.

Sorry, I'm not sure if I explained that well. Still, I'm open to any suggestions!

Reply:
You're right. Hmm. Interesting problem.

@RandellDawson Do you have any ides how we should go by tackling this?

Reply:
I need to review it again.  Give me some time.